### ISSUE TITLE
Support for chained conditional expressions

### ISSUE THREAD

**Issue by jnyrup** [fluentassertions/fluentassertions.analyzers#24]
Given a class like `MyClass`

```csharp
class MyClass
{
	public string MyProperty { get; set; }
}
```

The following tests all pass because they use a conditional operator before `Should()`.

```csharp
public void MyTestMethod0()
{
	var myClass = new MyClass();
	myClass.MyProperty?.Should().Be("foo");
}

[TestMethod]
public void MyTestMethod1()
{
	MyClass myClass = null;
	myClass?.MyProperty.Should().Be("foo");
}

[TestMethod]
public void MyTestMethod2()
{
	var myClass = new MyClass();
	myClass.MyProperty?[0].Should().Be('f');
}

[TestMethod]
public void MyTestMethod3()
{
	MyClass myClass = null;
	myClass?.MyProperty[0].Should().Be('f');
}
```

It would be nice if the analyzer from #22 could be extended to support these cases too.

I really really like that analyzer as it not only improves the assertions but catches bugs.
Awesome job!

**Comment by Meir017** [comment]
So basically using a null conditional expression together with the `Should` method should trigger the analyzer?

I was thinking about adding a code-fix that will recommend removing the null conditional expression, does this make sense?

**Comment by jnyrup** [comment]
I think using conditional expressions on the left hand side of `Should` should trigger the analyzer.
The conditional expression makes the assertion unclear - at least to me.

I had a unit test the other day like this
```csharp
actual?.Property.Should().NotBeNull();
```

where I think the intention was to check that both `actual` and `Property` should not be `null`.

So I rewrote it into
```csharp
actual.Should().NotBeNull();
actual.Property.Should().NotBeNull();
```

So I can't come up with a good code-fix, I guess it depends whether you would just remove the `?`, split it into two assertions or something else.

**Comment by Meir017** [comment]
ok, so, for now, I will just add support for a smarter analyzer without a code fix

**Comment by Meir017** [comment]
added more tests for NullConditionalAssertionAnalyzer 6b87da05061ea8ea6e47169b3fc9492a9c4f8e38
